### PR TITLE
Stats: Add Mobile Apps promo card via DotPager UI, take 2

### PR DESCRIPTION
**Pull request by a8ck3n**
#### Proposed Changes

Adds the Jetpack mobile apps promo card alongside the Yoast SEO one using the `DotPager` component. The Yoast card is still the default.

Before:

APromoSEO

After:

NewPromoCards

#### Testing Instructions

1. Visit the Calypso Live link.
2. Visit the Stats page.
3. Scroll down and confirm the `DotPager` card is present.
4. Confirm the Yoast SEO card is the default. 
5. Confirm the mobile apps card is available on the 2nd page.
6. Confirm it behaves appropriately on smaller screen sizes.
7. Test on a self-hosted Jetpack site and confirm the Yoast card and the `DotPager` UI are not used. Only the Jetpack mobile apps CTA should be shown.

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #69230 and supersedes #70272.

Currently just updating the Stats page. The designs call for the same promo card on the Insights and Ads pages but it looks overpowering there with the older page layout. I'm thinking we should wait until we "grid" those layouts before adding this component in.

### CODE REVIEW

**Review by keoshi — APPROVED** [review]
Looks and works great in my testing. Thanks for being accommodating and making the adjustments we discussed! 🙌

**Comment by jeffgolenski** [comment]
Looking and working great @a8ck3n! Only feedback / questions:
* I don't think "Jetpack" in the headline needs to be a link, since the description has the directions and same link. No need for the redundancy there!
* Just confirming: Do we have tracks events set for the link(s) and QR code set for the Jetpack card here? (I know yoast does!)
Thanks my friend!

**Comment by worldomonation** [comment]
I've re-triggered the failed tests. They both appear to be caused by flaky issues.

FYI, this link may be of interest to avoid review bomb: p7H4VZ-3Oh-p2#comment-10717

**Comment by a8ck3n** [comment]
> I don't think "Jetpack" in the headline needs to be a link, since the description has the directions and same link. No need for the redundancy there!

I was wondering about this myself. I'll change that up.

> Just confirming: Do we have tracks events set for the link(s) and QR code set for the Jetpack card here? (I know yoast does!)

No Tracks events. I'll look into adding that today.

**Comment by a8ck3n** [comment]
> FYI, this link may be of interest to avoid review bomb: p7H4VZ-3Oh-p2#comment-10717

Thanks for the link @worldomonation. I'll definitely review that and keep it handy for the future!

**Comment by a8ck3n** [comment]
Holding this until #70521 is merged as we want to wait until Tracks events are in.

**Comment by a8ck3n** [comment]
Closed in favour of #71118. With the Odyssey Stats changes in place, it was easier to apply these changes to a fresh checkout of trunk.

Note: All feedback in this PR has been addressed in the new PR.

**Pull request closed without merging**
